protocol/src/lib.rs

```diff
@@ -615,6 +617,104 @@ pub enum Message {
     Unknown(u32, BytesMut),
 }
 
+#[derive(Debug, PartialEq, Clone, Serialize, Deserialize)]
+pub struct WriteHeader {
+    pub upstairs_id: Uuid,
+    pub session_id: Uuid,
+    pub job_id: JobId,
+    pub dependencies: Vec<JobId>,
+    pub blocks: Vec<WriteBlockMetadata>,
+}
+
+impl WriteHeader {
+    /// Destructures into a list of block-size writes
+    ///
+    /// # Panics
+    /// `buf.len()` must be an even multiple of `self.blocks.len()`, which is
+    /// assumed to be the block size.
+    pub fn into_writes(&self, mut buf: bytes::Bytes) -> Vec<Write> {
```

Comment:
Sure, renamed to `get_writes` in 6ae04e89ddf5c1238b3fc2303ce7c67104919762

In the long term, I'd like to eliminate the `Vec<Write>` entirely from the Downstairs, since it's another unnecessary `pwritev` that could be a `pwrite`, but this PR was long enough already.